src/runtime/GlobalObjectBuiltinTypedArray.cpp

```diff
@@ -335,41 +364,39 @@ Value builtinTypedArrayConstructor(ExecutionState& state, Value thisValue, size_
         const Value& val = argv[0];
         if (!val.isObject()) {
             // $22.2.1.2 %TypedArray%(length)
-            int numlen = val.toNumber(state);
-            int elemlen = val.toLength(state);
-            if (numlen != elemlen)
-                ErrorObject::throwBuiltinError(state, ErrorObject::TypeError, state.context()->staticStrings().TypedArray.string(), false, String::emptyString, ErrorObject::Messages::GlobalObject_FirstArgumentInvalidLength);
+            double elemlen = val.toIndex(state);
```

Comment:
Return type of `toIndex` is "uint64_t", so please change the type of elemlen to "uint64_t" (operations for double type are generally expensive than integer type)